stem/IO/kratos_model_io.py

```diff
@@ -0,0 +1,547 @@
+            raise ValueError(f"Model part {body_model_part.name} has not been meshed."
+                             f"Before creating the mdpa file, the model part needs to be meshed."
+                             f"Please run Model.generate_mesh()")
+
+        # initialise block
+        block_text = ["", f"Begin Elements {kratos_element_type}"]
+        block_text.extend(
+            [self.__write_element_line(mat_id, el) for el in body_model_part.mesh.elements]
+        )
+        block_text += [f"End Elements", ""]
+        return block_text
+
+    def write_conditions_process_model_part(self, process_model_part: ModelPart, mat_id:int,
+                                            kratos_element_type:str):
+        """
+        Writes the conditions of the process model part to the mdpa file.
+
+        Args:
+            - process_model_part (:class:`stem.model_part.ModelPart`): the process model part to write to mdpa.
+            - mat_id (int): the material id connected to the conditions block
+            - kratos_element_type (str): the kratos element type for the condition.
+
+        Raises:
+            - ValueError: if model part is not a process model part
+            - ValueError: if mesh not initialised first
+
+        Returns:
+            - block_text (List[str]): list of strings for the elements of the body model part. \
+                Each element is a line in the mdpa file.
+        """
+        # validate part is body model part
+        if self.__is_body_model_part(process_model_part):
+            raise ValueError(f"Model part {process_model_part.name} is not a process model part!")
+
+        # check if mesh is initialised
+        if process_model_part.mesh is None:
+            raise ValueError(f"Model part {process_model_part.name} has not been meshed."
+                             f"Before creating the mdpa file, the model part needs to be meshed."
+                             f"Please run Model.generate_mesh()")
+
+        # no elements to write to conditions or process doesn't write condition elements
+        if process_model_part.mesh.elements is None or not self.__check_if_process_writes_conditions(
+                process_model_part
+        ):
+            block_text = []
+        else:
+            block_text = ["", f"Begin Conditions {kratos_element_type}"]
+            block_text.extend(
+                [self.__write_element_line(mat_id, el) for el in process_model_part.mesh.elements]
+            )
+            block_text += [f"End Conditions", ""]
+        return block_text
+
+    def __write_all_nodes(self, model):
+        """
+        Writes nodes to mdpa format.
+
+        Args:
+            - model (:class:`stem.model.Model`): the model object containing the info on the nodes.
+
+        Returns:
+            - block_text (List[str]): list of strings for the mdpa file. Each element is a line in the mdpa file.
+        """
+        nodes_dict = model.get_all_nodes()
```

Comment:
get_all_nodes is fine for now, but I want to change this in the future, to the same structure as Geometry.points